### DISCUSSION TITLE
@layer directive issues

### DISCUSSION THREAD

**Discussion by ciulic** [tailwindlabs/tailwindcss#4466] [Help] [answered]
https://tailwindcss.com/docs/functions-and-directives#layer

My problem is that I don't understand what layer allows me to do.
The documentation is talking about how you can leverage the directive, but how I can use what I have created I really can't understand from the docs.
I have these issue in React with css module context.

`@layer components {
  .my-btn-blue {
    @apply bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded;
  }
}`
How can I use my-btn-blue?
@apply btn-blue doesn't work, it fails with this error:
`uifym_1  | (7:32) /app/chameleon/ui-stack/ui/sidebar/nav-parent-item/styles.css The `my-btn-blue` class does not exist. If you're sure that `my-btn-blue` exists, make sure that any `@import` statements are being properly processed before Tailwind CSS sees your CSS, as `@apply` can only be used for classes in the same CSS tree.
uifym_1  |
uifym_1  |   5 | .link {
uifym_1  |   6 |     @apply flex items-center justify-between text-default font-medium text-gray-900;
uifym_1  | > 7 |     @apply hover:text-blue-600 my-btn-blue;
uifym_1  |     |                                ^
uifym_1  |   8 |
uifym_1  |   9 |     line-height: 2.4rem;`

And applying the class to an HTML element it generates no styles (due to css modules)
<div className="my-btn-blue" />

Any ideas what can I do?

**Comment by longtimedeveloper** [comment, marked as answer]
@ciulic 

This is how I've got my project using `@layer`

I hope this helps,

Karl

**SCSS**
```
@layer components {
  .container {
    @apply mx-auto w-80 sm:w-640 md:w-750 lg:w-970 xl:w-1170;
  }
}
```

**HTML** 
```
<div class="container">

</div>
```

**Reply by ciulic** [reply]
Thanks.
